irohad/ametsuchi/impl/postgres_wsv_common.hpp

```diff
@@ -106,43 +80,81 @@ namespace iroha {
       }
     }
 
+    template <typename Function, typename ParamType>
+    void processSOCI(soci::statement &st,
+                     soci::indicator &ind,
+                     ParamType &row,
+                     Function f) {
+      while (st.fetch()) {
+        switch (ind) {
+          case soci::i_ok:
+            f(row);
+            break;
+          case soci::i_null:
+          case soci::i_truncated:
+            break;
+        }
+      }
+    }
+
     static inline shared_model::builder::BuilderResult<
         shared_model::interface::Account>
-    makeAccount(const pqxx::row &row) noexcept {
-      return tryBuild([&row] {
+    makeAccount(const std::string &account_id,
+                const std::string &domain_id,
+                const shared_model::interface::types::QuorumType &quorum,
+                const std::string &data) noexcept {
+      return tryBuild([&] {
         return shared_model::builder::DefaultAccountBuilder()
-            .accountId(row.at("account_id").template as<std::string>())
-            .domainId(row.at("domain_id").template as<std::string>())
-            .quorum(
-                row.at("quorum")
-                    .template as<shared_model::interface::types::QuorumType>())
-            .jsonData(row.at("data").template as<std::string>())
+            .accountId(account_id)
+            .domainId(domain_id)
+            .quorum(quorum)
+            .jsonData(data)
             .build();
       });
     }
 
     static inline shared_model::builder::BuilderResult<
         shared_model::interface::Asset>
-    makeAsset(const pqxx::row &row) noexcept {
-      return tryBuild([&row] {
+    makeAsset(const std::string &asset_id,
+              const std::string &domain_id,
+              const int32_t precision) noexcept {
+      return tryBuild([&] {
         return shared_model::builder::DefaultAssetBuilder()
-            .assetId(row.at("asset_id").template as<std::string>())
-            .domainId(row.at("domain_id").template as<std::string>())
-            .precision(row.at("precision").template as<int32_t>())
+            .assetId(asset_id)
+            .domainId(domain_id)
+            .precision(precision)
             .build();
       });
     }
 
     static inline shared_model::builder::BuilderResult<
         shared_model::interface::AccountAsset>
-    makeAccountAsset(const pqxx::row &row) noexcept {
+    makeAccountAsset(const soci::row &row) noexcept {
```

Comment:
It's better to change the signature as well: pass 3 strings
Or maybe all similar function can use `soci::row`
Anyway that inconsistency should be resolved